### ISSUE TITLE
Concourse-Up failing to deploy to GCP

### ISSUE THREAD

**Issue by aguedes-pivotal** [EngineerBetter/concourse-up#87]
Hi,

I'm trying to use Concourse-UP to deploy Concourse to GCP with the following command:
`GOOGLE_APPLICATION_CREDENTIALS=service-account.json concourse-up deploy --iaas gcp project_name`

The Service account I have used is a Project Owner and I have checked and the CloudSQL APIs are enabled for this project, but I'm getting the following:

```
Error: Error applying plan:

1 error(s) occurred:

* google_sql_database_instance.director: 1 error(s) occurred:

* google_sql_database_instance.director: Error, failed to create instance terraform-20190206165509650100000001: googleapi: Error 400: Missing parameter: Tier., required

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


exit status 1
```

Thanks,
Alex

**Comment by DanielJonesEB** [comment]
Thanks for letting us know, @aguedes-pivotal. Are you using version 0.19.0, or did you build binary from source?

**Comment by aguedes-pivotal** [comment]
Thanks Dan, I’m using the 0.19.0 release.

On 6 Feb 2019, at 17:34, Daniel Jones <notifications@github.com> wrote:

Thanks for letting us know, @aguedes-pivotal
<https://github.com/aguedes-pivotal>. Are you using version 0.19.0, or did
you build binary from source?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/EngineerBetter/concourse-up/issues/87#issuecomment-461113940>,
or mute the thread

**Comment by mdcarlson** [comment]
+1 on this error.  

(using service account with project owner role as workaround for least-privileged service account)

**Comment by DanielJonesEB** [comment]
Hi folks,

Thanks for taking the time to get in touch. I'm looking into this currently - I can't reproduce using our CI service account, so investigating with a brand new account.

**Comment by DanielJonesEB** [comment]
When you first deployed, did you bump into a series of API enablement errors?

I've managed to reproduce with a fresh project where various APIs needed to be enabled. I then did a `concourse-destroy --iaas gcp my-deployment`, redeployed, and it seems to have got past this point.

I'd suspect that the API error is getting in the way of the Cloud SQL tier being set properly.

If you could confirm that you can work around this be doing a destroy and then another deploy, I'll raise stories about documenting the required APIs, and then close this issue.

**Comment by vchrisb** [comment]
I faced the same issue. When adding the db size parameter `--db-size small`, it works. It looks like a default is missing.

**Comment by DanielJonesEB** [comment]
@vchrisb Did you encounter any API enablement errors prior to the `Missing parameter: Tier.` error?

**Comment by vchrisb** [comment]
no, API was already enabled.

**Comment by vchrisb** [comment]
I went with the `--db-size` parameter, because the error suspects, that the `Tier` is simply not set.

**Comment by mdcarlson** [comment]
DJ, 

I used the destroy/re-deploy workaround and the `--region` flag to successfully deploy Concourse-Up on GCP in the `us-central1` region.

I did not have to specify a. `--db-size` flag.

The command I used was:
`concourse-up deploy --iaas gcp --region us-central1 snowflake`

-M

**Comment by DanielJonesEB** [comment]
Thanks for the confirmation, @mdcarlson.

I believe we've identified the cause of the bug - it should only affect failed deployments that are then resumed.

**Comment by evadinckel** [comment]
Hi,
This issue is now fixed in release 0.20.0 (https://github.com/EngineerBetter/concourse-up/releases/tag/0.20.0)
